pkg/agent/datamodel/sig_config.go

```diff
@@ -12,6 +16,151 @@ const (
 	AzurePublicCloudSigSubscription string = "109a5e88-712a-48ae-9078-9ca8b3c81345" // AKS VHD
 )
 
+//nolint:gochecknoglobals
+var (
+	CachedFromComponents = make(map[string]ContainerImage)
+	CachedFromManifest   = make(map[string]ProcessedManifest)
+)
+
+//nolint:gochecknoinits
+func init() {
+	manifest := CacheManifest()
+	processManifest(manifest)
+
+	components := CacheComponents()
+	processComponents(components)
+}
+
+func CacheManifest() Manifest {
+	_, filename, _, _ := runtime.Caller(0)
+	manifestFilePath := "../../../parts/linux/cloud-init/artifacts/manifest.json"
+	return getCachedVersionsFromManifestJSON(path.Join(path.Dir(filename), manifestFilePath))
+}
+
+func CacheComponents() Components {
+	_, filename, _, _ := runtime.Caller(0)
+	componentsFilePath := "../../../vhdbuilder/packer/components.json"
+	return getCachedVersionsFromComponentsJSON(path.Join(path.Dir(filename), componentsFilePath))
+}
+
+func processManifest(manifest Manifest) {
+	CachedFromManifest["kubernetes"] = ProcessedManifest{
+		Versions: manifest.Kubernetes.Versions,
+	}
+	CachedFromManifest["runc"] = ProcessedManifest{
+		Versions:  manifest.Runc.Versions,
+		Pinned:    manifest.Runc.Pinned,
+		Installed: manifest.Runc.Installed,
+	}
+	CachedFromManifest["containerd"] = ProcessedManifest{
+		Versions: manifest.Containerd.Versions,
+		Pinned:   manifest.Containerd.Pinned,
+		Edge:     manifest.Containerd.Edge,
+	}
+	CachedFromManifest["nvidia-container-runtime"] = ProcessedManifest{
+		Versions: manifest.NvidiaContainerRuntime.Versions,
+	}
+	CachedFromManifest["nvidia-drivers"] = ProcessedManifest{
+		Versions: manifest.NvidiaDrivers.Versions,
+	}
+}
+
+func processComponents(components Components) {
+	for _, image := range components.ContainerImages {
+		componentName, err := processContainerImageTag(image.DownloadURL)
+		if err != nil {
+			panic(err)
+		}
+		processed := ContainerImage{
+			MultiArchVersions: image.MultiArchVersions,
+			Amd64OnlyVersions: image.Amd64OnlyVersions,
+		}
+		if len(image.PrefetchOptimizations) > 0 {
+			processed.PrefetchOptimizations = PrefetchOptimizations{
+				Version:  image.PrefetchOptimizations[0].Version,
+				Binaries: image.PrefetchOptimizations[0].Binaries,
+			}
+		}
+		CachedFromComponents[componentName] = processed
+	}
+}
+
+func getCachedVersionsFromManifestJSON(manifestFilePath string) Manifest {
+	data, err := os.ReadFile(manifestFilePath)
+	if err != nil {
+		panic(err)
+	}
+	data = bytes.ReplaceAll(data, []byte("#EOF"), []byte(""))
+	var manifest Manifest
+	if err = json.Unmarshal(data, &manifest); err != nil {
+		panic(err)
+	}
+	return manifest
+}
+
+func getCachedVersionsFromComponentsJSON(componentsFilePath string) Components {
+	data, err := os.ReadFile(componentsFilePath)
+	if err != nil {
+		panic(err)
+	}
+	var components Components
+	if err = json.Unmarshal(data, &components); err != nil {
+		panic(err)
+	}
+	return components
+}
+
+type Manifest struct {
```

Comment:
I know I mentioned just doing raw JSON before but I think what you have is fine for now, can always revisit later